//----------------------------------------------------------------------------
// Copyright 2021, Ed Keenan, all rights reserved.
//----------------------------------------------------------------------------

syntax = "proto3";

import "vboData.proto";

option optimize_for = LITE_RUNTIME;

message meshData_proto
{
	enum RENDER_MODE
	{
		MODE_POINTS			= 0;
		MODE_LINE			= 1;
		MODE_LINE_LOOP		= 2;
		MODE_LINE_STRIP     = 3;
		MODE_TRIANGLES      = 4;
		MODE_TRIANGLE_STRIP = 5;
		MODE_TRIANGLE_FAN   = 6;
	};

	bool			enabled				= 1;
	bytes			pName				= 2;
	RENDER_MODE		mode				= 3;
	vboData_proto	vbo_vert			= 4;
	vboData_proto	vbo_norm			= 5;
	vboData_proto	vbo_uv				= 6;
	vboData_proto	vbo_index			= 7;
	vboData_proto	vbo_color			= 8;
	uint32			triCount			= 9;
	uint32			vertCount			= 10;
	uint32			materialIndex		= 11;
	sint32			jointIndex			= 12;
	sint32			parentJointIndex	= 13;
}													 
// -- End of File ---
